public/app/features/alerting/unified/components/rule-editor/TemplateStep/TemplateStep.tsx

```diff
@@ -0,0 +1,236 @@
+import { AlertRuleTemplateService } from 'app/percona/integrated-alerting/components/AlertRuleTemplate/AlertRuleTemplate.service';
+import {
+  Template,
+  TemplateParamType,
+} from 'app/percona/integrated-alerting/components/AlertRuleTemplate/AlertRuleTemplate.types';
+
+import { RuleFormValues } from '../../../types/rule-form';
+import { RuleEditorSection } from '../RuleEditorSection';
+
+import { AdvancedRuleSection } from './AdvancedRuleSection/AdvancedRuleSection';
+import TemplateFiltersField from './TemplateFiltersField';
+import { SEVERITY_OPTIONS, MINIMUM_DURATION_VALUE } from './TemplateStep.constants';
+import { Messages } from './TemplateStep.messages';
+import { AddAlertRuleModalService } from './TemplateStep.service';
+import { formatChannelsOptions, formatTemplateOptions } from './TemplateStep.utils';
+
+export const TemplateStep: FC = () => {
+  const {
+    register,
+    setValue,
+    formState: { errors },
+  } = useFormContext<RuleFormValues>();
+  const [templateOptions, setTemplateOptions] = useState<Array<SelectableValue<Template>>>();
+  const [loadingTemplates, setLoadingTemplates] = useState(false);
+  const [channelsOptions, setChannelsOptions] = useState<Array<SelectableValue<string>>>();
+  const [loadingChannels, setLoadingChannels] = useState(false);
+  const templates = useRef<Template[]>([]);
+  const [currentTemplate, setCurrentTemplate] = useState<Template>();
+  const [queryParams] = useQueryParams();
+  /* eslint-disable-next-line @typescript-eslint/consistent-type-assertions */
+  const selectedTemplate: string | null = (queryParams['template'] as string | undefined) || null;
+
+  const handleTemplateChange = useCallback(
+    (selectedTemplate?: Template, onChange?: (template?: Template) => void) => {
+      const newTemplate = templates.current.find((template) => template.name === selectedTemplate?.name);
+      const newDuration = newTemplate?.for;
+      const severityStr = newTemplate?.severity;
+      const newSeverity = SEVERITY_OPTIONS.find((severity) => severity.value === severityStr);
+
+      setCurrentTemplate(newTemplate);
+      if (newSeverity && newSeverity.value) {
+        // @ts-ignore
+        setValue('severity', newSeverity.value);
+      }
+      setValue('duration', parseInt(newDuration || '0', 10));
+
+      if (newTemplate) {
+        newTemplate.params?.forEach(({ type, float, name }) => {
+          // TODO add missing types when supported
+          if (type === TemplateParamType.FLOAT && float?.default !== undefined) {
+            // @ts-ignore
+            setValue(name, float.default);
+          }
+        });
+      }
+
+      if (!!onChange) {
+        onChange(selectedTemplate);
+      }
+    },
+    [setValue]
+  );
+
+  const getData = useCallback(async () => {
+    try {
+      setLoadingTemplates(true);
+      setLoadingChannels(true);
+      const [channelsListResponse, templatesListResponse] = await Promise.all([
```

Comment:
why don't we use dispatch for this? it seems to me if we took it out to the store it would look more neat in the component, what do you think?

Reply:
Very good point :D
I noticed they have an action for the channels, so I used it there. Then I sent the templates to the store. A lot cleaner now :)